cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1185,8 +1189,14 @@ private void deleteApp(ApplicationId appId, ApplicationSpecification spec) throw
     deleteAppMetadata(appId, spec);
     store.deleteWorkflowStats(appId);
     deletePreferences(appId, spec);
+
+    store.markDeleteApplication(appId);
+    // TODO : remove this in CDAP-20333 to test end to end delete
```

Comment:
why is this a todo instead of removing it now?

Reply:
need code changes to some functions like - getApplication, getLatestApplication etc (to filter out DELETED status), and refactoring tests which would mean a bigger PR - so I wanted to introduce the change incrementally

Reply:
There is no point right now, but will matter when we remove L1252, do you think we should introduce that in the subsequent PR?